src/components/fx/helpers.js

```diff
@@ -100,22 +100,58 @@ exports.appendArrayPointValue = function(pointData, trace, pointNumber) {
 
     for(var i = 0; i < arrayAttrs.length; i++) {
         var astr = arrayAttrs[i];
-        var key;
+        var key = getPointKey(astr);
 
-        if(astr === 'ids') key = 'id';
-        else if(astr === 'locations') key = 'location';
-        else key = astr;
+        if(pointData[key] === undefined) {
+            var val = Lib.nestedProperty(trace, astr).get();
+            var pointVal = getPointData(val, pointNumber);
+
+            if(pointVal !== undefined) pointData[key] = pointVal;
+        }
+    }
+};
+
+exports.appendArrayPointValues = function(pointData, trace, pointNumbers) {
```

Comment:
Maybe `appendArrayMultiPointValue` or `appendArrayMultiPointValues` would more typo-proof.

Reply:
... and please add jsDOC :arrow_heading_up:

Reply:
good call on `appendArrayMultiPointValues ` -> with docs in 
https://github.com/plotly/plotly.js/pull/2117/commits/4ba02497d920b1ff77cb0c878aff8209120104e1